app/jobs/delete_old_guests_job.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+class DeleteOldGuestsJob < ApplicationJob
+  non_tenant_job
+  after_perform do |_job|
+    reenqueue
+  end
+
+  def perform
+    User.where("guest = ? and updated_at < ?", true, Time.current - 7.days).each(&:destroy)
```

Comment:
Does the version of Rails you're targetting allow you to use `destroy_all` here?

Reply:
great question @prdanelli I'm working on fixing the broken spec right now. I'll try it out.  I admit copied the blacklight code from its rake task and didn't look at it too closely.